terraform/modules/rds/database.tf

```diff
@@ -38,6 +38,37 @@ resource "aws_db_instance" "this" {
   }
 }
 
+#tfsec:ignore:aws-rds-enable-performance-insights:TODO CLDC-2660 if necessary
+resource "aws_db_instance" "replica" {
+  #checkov:skip=CKV_AWS_129:cloudwatch logs TODO CLDC-2660
+  #checkov:skip=CKV_AWS_118:monitoring TODO CLDC-2660
+  #checkov:skip=CKV_AWS_353:performance insights TODO CLDC-2660 if necessary
+  #checkov:skip=CKV_AWS_354:performance insights TODO CLDC-2660 if insights are necessary
+  count = var.highly_available ? 1 : 0
+
+  identifier                 = "${var.prefix}-replica"
+  apply_immediately          = aws_db_instance.this.apply_immediately
+  auto_minor_version_upgrade = aws_db_instance.this.auto_minor_version_upgrade
+  copy_tags_to_snapshot      = aws_db_instance.this.copy_tags_to_snapshot
+  delete_automated_backups   = aws_db_instance.this.delete_automated_backups
+  deletion_protection        = true
+  final_snapshot_identifier  = aws_db_instance.this.final_snapshot_identifier
+  instance_class             = aws_db_instance.this.instance_class
+  maintenance_window         = aws_db_instance.this.maintenance_window
+  multi_az                   = aws_db_instance.this.multi_az
+  port                       = aws_db_instance.this.port
+  publicly_accessible        = aws_db_instance.this.publicly_accessible
+  replicate_source_db        = aws_db_instance.this.identifier
```

Comment:
The replica uses the same `aws_db_instance` terraform resource as the primary but also defines a `replicate_source_db`. Some attributes can't / shouldn't be set as per the [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance) and things I discovered during implementation, but the attributes that can be set I've opted to ensure are the same as the primary so that we have a true replica 😀